base/MAPL_SwathGridFactory.F90

```diff
@@ -0,0 +1,1259 @@
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+
+      _UNUSED_DUMMY(unusable)
+      grid = this%create_basic_grid(rc=status)
+      _VERIFY(status)
+
+      call this%add_horz_coordinates_from_file(grid, rc=status)
+      _VERIFY(status)
+
+      _RETURN(_SUCCESS)
+
+   end function make_new_grid
+
+
+   function create_basic_grid(this, unusable, rc) result(grid)
+      type (ESMF_Grid) :: grid
+      class (SwathGridFactory), intent(in) :: this
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+
+      _UNUSED_DUMMY(unusable)
+      
+      grid = ESMF_GridCreateNoPeriDim( &
+           & name = this%grid_name, &
+           & countsPerDEDim1=this%ims, &
+           & countsPerDEDim2=this%jms, &
+           & indexFlag=ESMF_INDEX_DELOCAL, &
+           & coordDep1=[1,2], &
+           & coordDep2=[1,2], &
+           & coordSys=ESMF_COORDSYS_SPH_RAD, &
+           & rc=status)
+      _VERIFY(status)
+      
+      ! Allocate coords at default stagger location
+      call ESMF_GridAddCoord(grid, rc=status)
+      _VERIFY(status)
+      call ESMF_GridAddCoord(grid, staggerloc=ESMF_STAGGERLOC_CORNER, rc=status)
+      _VERIFY(status)
+
+      if (this%lm /= MAPL_UNDEFINED_INTEGER) then
+         call ESMF_AttributeSet(grid, name='GRID_LM', value=this%lm, rc=status)
+         _VERIFY(status)
+      end if
+
+      call ESMF_AttributeSet(grid, 'GridType', 'LatLon', rc=status)   ! grid=ESMF_grid
+      _VERIFY(status)
+
+      call ESMF_AttributeSet(grid, 'Global', .false., rc=status)
+
+
+      _RETURN(_SUCCESS)
+   end function create_basic_grid
+
+   
+
+   subroutine add_horz_coordinates_from_file(this, grid, unusable, rc)
+      use MAPL_BaseMod, only: MAPL_grid_interior
+      class (SwathGridFactory), intent(in) :: this
+      type (ESMF_Grid), intent(inout) :: grid
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      integer, optional, intent(out) :: rc
+
+      integer :: i_1, i_n, j_1, j_n ! regional array bounds
+      integer :: ic_1,ic_n,jc_1,jc_n ! regional corner bounds
+
+      !! shared mem
+      real(kind=ESMF_KIND_R8), pointer :: fptr(:,:)
+      real, pointer :: centers(:,:)               ! check R4 or R8
+
+      integer :: status
+      integer :: i, j, ij(4)
+      integer :: Xdim, Ydim
+      
+      integer :: IM, JM
+      integer :: IM_WORLD, JM_WORLD
+      integer :: COUNTS(3), DIMS(3)
+      character(len=:), allocatable :: lon_center_name, lat_center_name
+      
+      
+      _UNUSED_DUMMY(unusable)
+
+      if (this%initialized_from_metadata) then
+         lon_center_name = "lons"
+         lat_center_name = "lats"
+      else
+         ! keywords in netCDF
+         lon_center_name = "lon_centers"
+         lat_center_name = "lat_centers"
+      end if
+      
+      call MAPL_grid_interior(grid, i_1, i_n, j_1, j_n)
+      !!write(6,*) 'i_1, i_n, j_1, j_n', i_1, i_n, j_1, j_n
```

Comment:
Delete dead code.

@metdyn  When the dust settles, we can show you how to use pFlogger to add debug messages that will be suppressed in typical runs.  This allows us to turn on such diagnostics without recompiling.

Reply:
I have deleted this line and those similar lines where I used to debug.